FreeRTOS/FreeRTOS-Kernel
# Replace sprintf with snprintf

Description
-----------
This change necessitates the introduction of 2 new APIs:

```c
void vTaskListTasks( char * pcWriteBuffer, size_t uxBufferLength );
void vTaskGetRunTimeStatistics( char * pcWriteBuffer, size_t uxBufferLength );
```

These 2 APIs behave exactly as `vTaskList` and `vTaskGetRunTimeStats` except the fact that they take the length of the `pcWriteBuffer` as the second argument to ensure that we do not write past the buffer.

`vTaskList` and `vTaskGetRunTimeStats` assume that the length of the buffer is `configSTATS_BUFFER_MAX_LENGTH` which defaults to `0xFFFF`. This is done to ensure that the existing applications do not break. New applications should use the new APIs to avoid memory corruption.

Test Steps
-----------
Tested behavior of these APIs for various buffer sizes on Cortex-M7.

Checklist:
----------



Related Issue
-----------
* https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/746


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
